OneCommunityGlobal/HighestGoodNetworkApp
# Nathan highlight bio permission

# Description
This PR adds a permission for highlighting the "Bio announcement" row on the Weekly Summaries Report page given certain criteria. 


## Main changes explained:
- Add `highlightEligibleBios` to `PermissionsConst.js`
- Check for `highlightEligibleBios` in `WeeklySummariesReport.jsx`
- Move highlight logic so it's not just for the bio toggle component

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log in as an owner and give volunteer accounts the `highlightEligibleBios` permission labeled "See Highlight for Bios Eligible to be Posted" in Permissions Management→ Manage User Permissions
5. log in as users with and without the permission
6. go to dashboard→ Reports→ Weekly Summaries Report
7. find `0Sherly Volunteer` and verify the `Bio announcement` row isn't highlighted when you don't have the permission
8. find `0Sherly Volunteer` and verify the `Bio announcement` row is highlighted when you t have the permission

## Screenshots or videos of changes:
No highlight:
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/328c5808-3757-4908-8fd4-5a46deaeb97c)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/2dffdcea-6c55-42e2-a81c-e2d6bc335a06)

Highlight: 
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/7e7ceee7-d92d-4c41-b0e5-cc8dc932deb8)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/23086054/d490b007-7db1-423b-a9aa-29f820614254)

## Note:
You can check what permission your user has with redux dev tools under:
1. `state.auth.user.permissions.frontPermissions` for individual permissions
2. `state.role.roles.{the role of your account}` for role permissions

If you need to assign permissions to a user, log in with an Owner account and go to Other Links->Permission Management->Manage User Permissions and give/remove the needed permissions to/from a volunteer account. You will have to log out and back in to update your permissions if you stayed logged in in another tab/window.
